File: cs/sample/Sample/FindLuckyIntInArr/FindLuckyIntInArr.cs
using System;
using System.Collections.Generic;

namespace Sample.FindLuckyIntInArr{
    public class findLuckyIntInArr{
        public static int naive(int[] arr){
            Dictionary<int,int> d = new Dictionary<int,int>();
            for (int i=0;i<arr.Length;i++)
                d[arr[i]] = d.ContainsKey(arr[i])?d[arr[i]]+1:1;
            int result = -1;
            foreach (var kvp in d){
                if (kvp.Key==kvp.Value){
                    if (kvp.Key > result)
                        result = kvp.Key;
                }
            }
            return result;
        }
    }
}

File: cpp/example/src/FindLuckyIntInArr/findLuckyIntInArr.cpp
#include "findLuckyIntInArr.h"

int findLuckyIntInArr::naive(vector<int>& arr){
    map<int,int> d = {};
    map<int,int>::iterator it;
    for (int i=0;i<arr.size();i++){
        it = d.find(arr[i]);
        d[arr[i]] = it!=d.end()?d[arr[i]]+1:1;
    }
    int result=-1;
    it = d.begin();
    while (it!=d.end()){
        if (it->first==it->second){
            if (it->first > result)
                result = it->first;
        }
        it++;
    }
    return result;
}
